src/sync/sync-utils.ts

```diff
@@ -65,25 +51,53 @@ export const findOrStartSync = async (
 	}, 0, logger);
 };
 
-type SubscriptionUpdateTasks = {
-	totalNumberOfRepos?: number | null;
-	repositoryCursor?: string | null;
-	repositoryStatus?: string | null;
-}
+const getCommitsFromDates = async (jiraHost: string, commitsFromDate: Date | undefined) => {
+	const mainCommitsFromDate = await getCommitSinceDate(jiraHost, NumberFlags.SYNC_MAIN_COMMIT_TIME_LIMIT, commitsFromDate?.toISOString());
+	const branchCommitsFromDate = await getCommitSinceDate(jiraHost, NumberFlags.SYNC_BRANCH_COMMIT_TIME_LIMIT, commitsFromDate?.toISOString());
+	return { mainCommitsFromDate, branchCommitsFromDate };
+};
 
-const resetTargetedTasks = async (subscription: Subscription, syncType?: SyncType, targetTasks?: TaskType[]): Promise<void> => {
-	if (!targetTasks?.length) {
-		return;
+const resetRepoTaskStatusAndCursor = async (subscription: Subscription, syncType: SyncType, targetTasks?: TaskType[]) => {
+
+	const repoSyncTasks = (targetTasks || []).filter(t => t !== "repository");
+	const targetTaskIsEmpty = (targetTasks || []).length === 0;
+	const hasSubscriptionRepositoryTask = (targetTasks || []).includes("repository");
+	const hasRepoSyncTask = repoSyncTasks.length > 0;
+
+	if (syncType == "full") {
+		//For full sync
+		if (targetTaskIsEmpty) {
+			// Didn't provide any target tasks for a full sync, so remove all state as we're starting anew
+			await subscription.update({ totalNumberOfRepos: null, repositoryCursor: null, repositoryStatus: null });
+			await RepoSyncState.deleteFromSubscription(subscription);
+		} else {
+			if (hasRepoSyncTask) {
+				await resetTargetedTasks({ subscription, repoSyncTasks, resetCursor: true });
+			}
+			if (hasSubscriptionRepositoryTask) {
+				await subscription.update({ totalNumberOfRepos: null, repositoryStatus: null, repositoryCursor: null });
+			}
+			await RepoSyncState.update({ failedCode: null }, { where: { subscriptionId: subscription.id } });
+		}
+	} else {
```

Comment:
should we maybe explcitly check that it is partial and throw an exception otherwise?

Reply:
I thought about that, and can do that, and remove the comments.